site/components/DateSelector.tsx

```diff
@@ -1,81 +1,100 @@
-                        type="text"
-                        defaultValue={inputs?.yearInput}
-                    />
-                </div>
+const inputBox = (
+    startOrEnd: "start" | "end",
+    hasErrors: boolean,
+    inputRef: React.Ref<any> | undefined,
+    inputProps: InputBaseComponentProps | undefined,
+    InputProps: Partial<FilledInputProps> | Partial<OutlinedInputProps> | undefined,
+    disabled: boolean,
+    inputId: string,
+) => (
+    <div className="govuk-date-input flex items-center [&_.MuiSvgIcon-root]:fill-govBlue" id={`${startOrEnd}-date`}>
+        <div className="govuk-date-input__item govuk-!-margin-right-0">
+            <div className="govuk-form-group">
+                <input
+                    className={`govuk-input govuk-date-input__input govuk-input--width-6 ${
+                        hasErrors ? "govuk-input--error" : ""
+                    } `}
+                    id={inputId}
+                    name={`${inputId.includes("publish") ? "publish" : "validity"}-${startOrEnd}DateDay`}
+                    type="text"
+                    ref={inputRef}
+                    {...inputProps}
+                    disabled={disabled}
+                />
             </div>
         </div>
-    </FormElementWrapper>
+        {InputProps?.endAdornment}
+    </div>
+);
+
+const renderWeekPickerDay = (
+    _date: Date,
+    _selectedDates: Array<Date | null>,
+    pickersDayProps: PickersDayProps<Date>,
+) => (
+    <PickersDay
+        {...pickersDayProps}
+        classes={{
+            selected: "!bg-govBlue",
+            dayWithMargin:
+                "focus:!border focus:!border-solid hover:!border hover:!border-solid hover:!border-govBlue focus:!border-govYellow",
+        }}
+    />
 );
 
+const DateSelector = ({
+    input = null,
+    startOrEnd,
+    errors = [],
+    disabled,
+    inputId,
+}: DateSelectorProps): ReactElement => {
+    const [value, setValue] = useState(input);
+
+    return (
+        <FormElementWrapper errors={errors} errorId={`${startOrEnd}-day-input`} errorClass="govuk-date-input--error">
+            <LocalizationProvider dateAdapter={AdapterDayjs}>
+                <DatePicker
+                    renderDay={renderWeekPickerDay}
+                    value={value}
+                    onChange={(newValue) => {
+                        setValue(newValue);
+                    }}
+                    renderInput={({ inputRef, inputProps, InputProps }) => {
+                        return inputBox(
+                            startOrEnd,
+                            errors.length > 0,
+                            inputRef,
+                            inputProps,
+                            InputProps,
+                            disabled,
+                            inputId,
+                        );
+                    }}
+                    disablePast={startOrEnd === "end"}
```

Comment:
If we pass `disablePast` as a prop then we can get rid of `startOrEnd`

Reply:
done